src/modules/peek/Peek.FilePreviewer/Previewers/UnsupportedFilePreviewer/UnsupportedFilePreviewer.cs

```diff
@@ -97,139 +102,61 @@ await Dispatcher.RunOnUiThread(async () =>
             });
         }
 
-        public async Task<bool> LoadIconPreviewAsync(CancellationToken cancellationToken)
+        private async Task LoadIconPreviewAsync(CancellationToken cancellationToken)
         {
-            bool isIconValid = false;
-
-            var isTaskSuccessful = await TaskExtension.RunSafe(async () =>
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-                await Dispatcher.RunOnUiThread(async () =>
-                {
-                    cancellationToken.ThrowIfCancellationRequested();
-
-                    var iconBitmap = await ThumbnailHelper.GetThumbnailAsync(Item.Path, cancellationToken)
-                        ?? await ThumbnailHelper.GetIconAsync(Item.Path, cancellationToken);
-
-                    cancellationToken.ThrowIfCancellationRequested();
-
-                    isIconValid = iconBitmap != null;
-
-                    Preview.IconPreview = iconBitmap ?? new SvgImageSource(new Uri("ms-appx:///Assets/Peek/DefaultFileIcon.svg"));
-                });
-            });
-
-            return isIconValid && isTaskSuccessful;
+            Preview.IconPreview = await ThumbnailHelper.GetThumbnailAsync(Item.Path, cancellationToken) ??
+                await ThumbnailHelper.GetIconAsync(Item.Path, cancellationToken) ??
+                DefaultIcon;
         }
 
-        public async Task<bool> LoadDisplayInfoAsync(CancellationToken cancellationToken)
+        private async Task LoadDisplayInfoAsync(IProgress<string?> sizeProgress, CancellationToken cancellationToken)
         {
-            bool isDisplayValid = false;
-
-            var isTaskSuccessful = await TaskExtension.RunSafe(async () =>
-            {
-                cancellationToken.ThrowIfCancellationRequested();
-
-                var type = await Task.Run(Item.GetContentTypeAsync);
-
-                cancellationToken.ThrowIfCancellationRequested();
-
-                isDisplayValid = type != null;
-
-                var readableFileSize = string.Empty;
+            string? type = await Item.GetContentTypeAsync();
```

Comment:
Can this be changed to `string` or `var`?

Reply:
Sure. Thanks 😊